### PR TITLE
Fix ServiceWorker registration failed

### PR DESCRIPTION
**Pull request by ghost**
Fixes: #24490 

I explain what I'm doing here:

1. I change the destination of 'workbox-sw' from **'assets/js/vendor'** to **'assets/js/sw'**. The change of destination is because I want to include the folder in **'keep_files'** key of **'_config.yml'**.

2. I include './sw.js' in the files to exclude during build, since this file only serves as a template (it should not be copied to gh_pages, as it would overwrite the correctly generated file).

Then:

1. After `docs-compile`, the workbox files are copied and the manifest is generated in gh_pages.
2. During `docs-serve`, Jekyll deletes all files, but keeps the files needed for the service worker.

error-sw-fixed

As the current Bootstrap documentation mentions, `npm run docs` must be run at least once for the service worker files to be generated.

### CODE REVIEW

File: build/workbox.js [hunk 1]
```diff
@@ -8,10 +8,12 @@ const buildPrefix = '_gh_pages/'
 
 const workboxSWSrcPath = require.resolve('workbox-sw')
 const wbFileName = path.basename(workboxSWSrcPath)
-const workboxSWDestPath = buildPrefix + 'assets/js/vendor/' + wbFileName
+const workboxSWDestDir = path.resolve(buildPrefix + 'assets/js/sw')
+const workboxSWDestPath = path.join(workboxSWDestDir, wbFileName)
 const workboxSWSrcMapPath = `${workboxSWSrcPath}.map`
 const workboxSWDestMapPath = `${workboxSWDestPath}.map`
 
+if (!fs.existsSync(workboxSWDestDir)) fs.mkdirSync(workboxSWDestDir)
```
Comment (XhmikosR) [inline build/workbox.js]: brackets and new lines

**Comment by XhmikosR** [comment]
Why can't we just keep the file in vendor?

**Comment by XhmikosR** [comment]
Also, with your solution, how will the precache files be updated during the serve command?

**Comment by lucascono** [comment]
Look at the changes @XhmikosR . This works both during `npm run docs-compile` and during `npm run docs-serve`.

**Comment by XhmikosR** [comment]
I'm not sure I like it. Tying things more to Jekyll. Plus the Travis build doesn't even run for your PR so I'll need to check it out locally.

At any rate, I think, I'd rather skip this kind of solution and just live with the local warning, until we have a proper solution.

**Pull request closed without merging**